### PR TITLE
Add ALTER TABLE UNFREEZE command.

### PR DESCRIPTION
**Pull request by Jokser**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added new SQL command ALTER TABLE 'table_name' UNFREEZE [PARTITION 'part_expr'] WITH NAME 'backup_name' 

Detailed description / Documentation draft:
This command is needed to properly remove 'freezed' partitions from all disks.

### CODE REVIEW

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -3727,6 +3735,20 @@ PartitionCommandsResultInfo MergeTreeData::freezePartitionsByMatcher(MatcherFn m
     return result;
 }
 
+PartitionCommandsResultInfo MergeTreeData::unfreeze(const String & backup_name, const Context & context, TableLockHolder &) const
+{
+    String clickhouse_path = Poco::Path(context.getPath()).makeAbsolute().toString();
+    String backup_path = "shadow/" + backup_name + "/" + relative_data_path;
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeData.cpp:3741]: Path traversal vulnerability.

Comment (alexey-milovidov) [reply src/Storages/MergeTree/MergeTreeData.cpp:3741]: Please also rewrite with std::filesystem.

File: tests/integration/test_merge_tree_s3/test.py [hunk 2]
```diff
@@ -333,3 +333,26 @@ def test_move_replace_partition_to_another_table(cluster):
...
+
+    node.query("ALTER TABLE s3_test FREEZE WITH NAME 'backup1'")
+    assert len(
+        list(minio.list_objects(cluster.minio_bucket, 'data/'))) == FILES_OVERHEAD + FILES_OVERHEAD_PER_PART_WIDE
+
+    node.query("TRUNCATE TABLE s3_test")
+    assert len(
+        list(minio.list_objects(cluster.minio_bucket, 'data/'))) == FILES_OVERHEAD + FILES_OVERHEAD_PER_PART_WIDE
+
+    node.query("ALTER TABLE s3_test UNFREEZE 'backup1'")
```
Comment (alexey-milovidov) [inline tests/integration/test_merge_tree_s3/test.py:356]: This command looks inconsistent.
Maybe better to make it look `UNFREEZE WITH NAME '...'`?
And also add `UNFREEZE PARTITION`?

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

**Review by alesapin — APPROVED** [review]
Need simple functional tests with the table in old syntax.

**Comment by Jokser** [comment]
@alesapin Could you please check the change again?

**Comment by alesapin** [comment]
No realted failures

**Pull request merged**